shm/cargo.cpp

```diff
@@ -0,0 +1,38 @@
+#include "cargo.hpp"
+
+#include <iostream>
+#include <string>
+
+Cargo::Cargo(std::string name, size_t amount, size_t basePrice)
+    : name_(name), amount_(amount), basePrice_(basePrice) {}
+
+Cargo& Cargo::operator+=(size_t amount) {
+    amount_ += amount;
+    return *this;
+}
+Cargo& Cargo::operator-=(size_t amount) {
+    if (amount_ < amount) {
+        std::cerr << "Cargo is lower than 0 \n";
```

Comment:
Just a question: Will `std:cerr` stop the program working? Maybe sufficient is to make simply std::clamp() or return `0` (I want to highlight: It is just a question ;p)

Reply:
Cerr won't stop the program, It will only says us that we can't go lower than 0, and don't modify value.
It's temporary solution.